test/repository/repository_spec.rb

```diff
@@ -0,0 +1,98 @@
+require "minitest_helper"
+
+describe Koine::Repository::Repository do
+  let(:adapter) { MiniTest::Mock.new }
+  let(:subject) { Koine::Repository::Repository.new(adapter) }
+
+  describe "#save" do
+    it "throws an exception" do
+      begin
+        subject.save(ArticleEntity.new)
+        fail
+      rescue RuntimeError => e
+        e.message.must_equal "Method not implemented"
+      end
+    end
+  end
+
+  describe "#remove" do
+    it "throws an exception" do
+      begin
+        subject.remove(ArticleEntity.new)
+        fail
+      rescue RuntimeError => e
+        e.message.must_equal "Method not implemented"
+      end
+    end
+  end
+
+  describe "#adapter" do
+    it "returns the adapter" do
+      adapter = []
+      klass = subject.class
+      subject = klass.new(adapter)
+      subject.send(:adapter).must_be_same_as(adapter)
+    end
+  end
+
+  describe "#hydrator" do
+    it "defaults to Koine::Hydrator::Hydrator" do
+      subject.hydrator.must_be_same_as(subject.hydrator)
+      subject.hydrator.must_be_instance_of(Koine::Hydrator::Hydrator)
+    end
+
+    it "can be mutated" do
+      hydrator = Koine::Hydrator::Hydrator.new
+      subject.hydrator = hydrator
+
+      subject.hydrator.must_be_same_as(hydrator)
+    end
+  end
+
+  describe "#entity_prototype" do
+    it "throws exception when no entity prototype was given" do
+      begin
+        subject.entity_prototype
+        fail
+      rescue RuntimeError => e
+        e.message.must_equal "Entity prototype was not set"
+      end
+    end
+
+    it "can be mutated" do
+      prototype = ArticleEntity.new
+      subject.entity_prototype = prototype
+      subject.entity_prototype.must_be_same_as(prototype)
+    end
+  end
+
+  describe "#find_all_by" do
+    it "returns a collection of entities" do
+      criterias = { foo: :bar };
+
+      collection = [
+        { id: 1, title: 'title 1', body: 'body 1' },
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.